stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/nativeimpl/channel/listener/Start.java

```diff
@@ -85,170 +82,124 @@ public void execute(Context context) {
         try {
-            DeliverCallback deliverCallback = (consumerTag, delivery) -> {
-                Executor.submit(resource, new ResponseCallback(), null, null,
-                        new BInteger(Integer.parseInt(new String(delivery.getBody(), RabbitMQConstants.CHAR_SET))));
-            };
-            channel.basicConsume(queue, true, deliverCallback, consumerTag -> {
-            });
+            channel.basicConsume(queueName, autoAck,
+                    new DefaultConsumer(channel) {
+                        @Override
+                        public void handleDelivery(String consumerTag,
+                                                   Envelope envelope,
+                                                   AMQP.BasicProperties properties,
+                                                   byte[] body) throws IOException {
+                            dispatchMessage(resource, body, channel, envelope.getDeliveryTag(), autoAck);
+                        }
+                    });
         } catch (IOException exception) {
             throw new BallerinaException(exception);
         }
     }
 
     /**
-     * Dispatch messages in float format.
+     * Create and get message BMap.
      *
      * @param resource Ballerina resource function.
-     * @param channel  RabbitMQ channel.
-     * @param queue    Name of the queue.
+     * @param message  Message content received from the RabbitMQ server.
+     * @param channel  RabbitMQ Channel object.
+     * @return Ballerina RabbitMQ message BValue.
      */
-    private void dispatchFloatMessage(Resource resource, Channel channel, String queue) {
-        try {
-            DeliverCallback deliverCallback = (consumerTag, delivery) -> {
-                Executor.submit(resource, new ResponseCallback(), null, null,
-                        new BFloat(Float.parseFloat(new String(delivery.getBody(), RabbitMQConstants.CHAR_SET))));
-            };
-            channel.basicConsume(queue, true, deliverCallback, consumerTag -> {
-            });
-        } catch (IOException exception) {
-            throw new BallerinaException(exception);
+    private BValue getMessageBMap(Resource resource, byte[] message, Channel channel, long deliveryTag,
+                                  boolean autoAck) {
+        ProgramFile programFile = resource.getResourceInfo().getPackageInfo().getProgramFile();
+        BMap<String, BValue> messageObj = BLangConnectorSPIUtil.createBStruct(
+                programFile, RabbitMQConstants.PACKAGE_RABBITMQ, RabbitMQConstants.MESSAGE_OBJECT);
+        messageObj.addNativeData(RabbitMQConstants.DELIVERY_TAG, deliveryTag);
+        messageObj.addNativeData(RabbitMQConstants.CHANNEL_NATIVE_OBJECT, channel);
+        messageObj.addNativeData(RabbitMQConstants.MESSAGE_CONTENT, message);
+        if (autoAck) {
+            messageObj.addNativeData(RabbitMQConstants.AUTO_ACK_STATUS, true);
         }
```

Comment:
Shall we remove the if clause and set the value like,
messageObj.addNativeData(RabbitMQConstants.AUTO_ACK_STATUS, autoAck);

Reply:
Resolved